### PR TITLE
Fix bugs related to track files

### PR DESCRIPTION
**Pull request by paulromano**
This pull request fixes a few issues I ran into recently when trying to generate a plot of neutron tracks from a fixed source problem:

- The `-t` command-line flag didn't work for fixed source problems.
- Creating particle track files would fail when threading was turned on. I've fixed this by adding `!$omp critical` around the file writing.
- The openmc-track-to-vtk and openmc-voxel-to-silovtk scripts (recently renamed) failed when trying to generate VTK files. The cause was a change in the VTK 6 API (vtkXMLImageDataWriter.SetInput --> vtkXMLImageDataWriter.SetInputData) as described [here](http://www.vtk.org/Wiki/VTK/VTK_6_Migration/Replacement_of_SetInput).
- I've also cleaned up those two scripts (PEP8 compliance) and made a few changes in the interest of being more Pythonic.

### CODE REVIEW

**Comment by paulromano** [comment]
@smharper I'll tentatively assign you as reviewer since you implemented the track file capability.

File: scripts/openmc-track-to-vtk [hunk 1]
```diff
@@ -65,7 +65,7 @@ def main():
         # Write coordinate values to points array.
         if fname.endswith('.binary'):
             track = open(fname, 'rb').read()
-            coords = [struct.unpack("ddd", track[24*i : 24*(i+1)])
+            coords = [struct.unpack("ddd", track[24*i:24*(i+1)])
```
Comment (smharper) [inline scripts/openmc-track-to-vtk]: I personally think this looks too cluttered without the whitespace, and the whitespace is allowed in PEP 8 (see the third bullet in https://www.python.org/dev/peps/pep-0008/#pet-peeves).

Comment (paulromano) [inline scripts/openmc-track-to-vtk]: You're right; I've added the spaces back.

File: scripts/openmc-track-to-vtk [hunk 2]
```diff
@@ -74,26 +74,25 @@ def main():
...
         cells.InsertNextCell(line)
         point_offset += n_points
     data = vtk.vtkPolyData()
     data.SetPoints(points)
     data.SetLines(cells)
-    
+
     writer = vtk.vtkXMLPPolyDataWriter()
-    writer.SetInput(data)
+    writer.SetInputData(data)
```
Comment (smharper) [inline scripts/openmc-track-to-vtk]: This doesn't work for me with my outdated VTK.  I think the version of VTK I have is the one you get with `sudo apt-get install python-vtk` which we should try to support.  What do you think about changing this to

```python
    if (int(vtk.vtkVersion.GetVTKVersion().split('.')[0]) > 5):
        writer.SetInputData(data)
    else:
        writer.SetInput(data)
```

Comment (smharper) [inline scripts/openmc-track-to-vtk]: I should specify that I am using Ubuntu 14.04

Comment (paulromano) [inline scripts/openmc-track-to-vtk]: Ah, ok. On 14.10, I guess they updated python-vtk to 6, hence the differences in APIs we're seeing. I used `vtk.vtkVersion.GetVTKMajorVersion()` to implement what you suggested, so now it should work for any version.

**Comment by smharper** [comment]
Do you know why the Travis build failed?

**Comment by smharper** [comment]
Everything looks good to me.  Thanks for the PR @paulromano.

**Pull request merged**